File: ueueo-framework/ueueo-authorization/src/main/java/com/ueueo/authorization/permissions/RequirePermissionsSimpleBatchStateCheckerModel.cs
using Volo.Abp.SimpleStateChecking;

namespace Volo.Abp.Authorization.Permissions;

public class RequirePermissionsSimpleBatchStateCheckerModel<TState>
    //where TState : IHasSimpleStateCheckers<TState>
{
    public TState State;//  { get; }

    public String[] Permissions;//  { get; }

    public boolean RequiresAll;//  { get; }

    public RequirePermissionsSimpleBatchStateCheckerModel(TState state, String[] permissions, boolean requiresAll = true)
    {
        Objects.requireNonNull(state, nameof(state));
        Check.NotNullOrEmpty(permissions, nameof(permissions));

        State = state;
        Permissions = permissions;
        RequiresAll = requiresAll;
    }
}

File: ueueo-framework/ueueo-authorization/src/main/java/com/ueueo/authorization/permissions/RequirePermissionsSimpleBatchStateCheckerModel.java
package com.ueueo.authorization.permissions;

import org.apache.commons.collections4.CollectionUtils;
import org.springframework.util.Assert;

import java.util.List;

public class RequirePermissionsSimpleBatchStateCheckerModel<TState> {

    public TState state;

    public List<String> permissions;

    public boolean requiresAll = true;

    public RequirePermissionsSimpleBatchStateCheckerModel(TState state, List<String> permissions, Boolean requiresAll) {
        Assert.notNull(state, "State must not null!");
        Assert.isTrue(CollectionUtils.isNotEmpty(permissions), "Permissions must not empty!");

        this.state = state;
        this.permissions = permissions;
        if (requiresAll != null) {
            this.requiresAll = requiresAll;
        }
    }

    public TState getState() {
        return state;
    }

    public List<String> getPermissions() {
        return permissions;
    }

    public boolean isRequiresAll() {
        return requiresAll;
    }
}
